### PR TITLE
[T3A3][T11-A2]Sun Mingyang

### PR DESCRIPTION
**Pull request by nightism**

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -25,7 +38,17 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        //this.value = address;
+        
+        this.addressComponents = address.split(", ");
+        if (!isValidAddress(this.addressComponents)) {
```
Comment (xdshang) [inline src/seedu/addressbook/data/person/Address.java]: It is better to merge the two validation methods. It doesn't make sense that it still could be invalid after the 1st validation test.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -9,12 +9,25 @@
...
-
-    public final String value;
+    
+    private static final int ADDRESS_INDEX_BLOCK = 0;
+    private static final int ADDRESS_INDEX_STREET = 1;
+    private static final int ADDRESS_INDEX_UNIT = 2;
+    private static final int ADDRESS_INDEX_POSTAL = 3;
+    
+    //public final String value;
+    public final String[] addressComponents;
```
Comment (xdshang) [inline src/seedu/addressbook/data/person/Address.java]: This variable only stores the intermediate results, so it is not good to be declared as a class member here,  but the inside the method that uses it.

**Comment by xdshang** [comment]
You should update the test cases once after you made change to your program.
Also, this PR contains too many extra files, e.g. .metadata/*. You should add them to `.gitignore`, so that they won't be staged.

**Comment by xdshang** [comment]
@nightism Some comments added. Please close the PR after reading comments.

**Comment by nightism** [comment]
test cases can fit the program now

**Pull request closed without merging**